### ISSUE TITLE
'unknown type name 'time64_t'' error for mingw-w64 builds on Windows

### ISSUE THREAD

**Issue by sav-ix** [openssl/openssl#3249]
Hi, everyone,

For OpenSSL builds using mingw-w64 on Windows got error:
```
sh-4.4$ ./Configure threads no-deprecated shared mingw64 >_Configure.log && make
[snip]
make depend && make _all
make[1]: Entering directory '/c/libOPENSSL-1.1.1-dev/build'
make[1]: Leaving directory '/c/libOPENSSL-1.1.1-dev/build'
make[1]: Entering directory '/c/libOPENSSL-1.1.1-dev/build'
gcc  -Iinclude -DDSO_WIN32 -DNDEBUG -DOPENSSL_THREADS -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DRC4_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPADLOCK_ASM -DPOLY1305_ASM -DOPENSSL_API_COMPAT=0x10100000L -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib/engines-1_1\"" -DL_ENDIAN -DWIN32_LEAN_AND_MEAN -DUNICODE -D_UNICODE -m64 -Wall -O3 -D_MT   -MMD -MF fuzz/client.d.tmp -MT fuzz/client.o -c -o fuzz/client.o fuzz/client.c
fuzz/client.c:40:1: error: unknown type name 'time64_t'
 time64_t _time64(time64_t *t) TIME_IMPL(t)
 ^~~~~~~~
fuzz/client.c:40:18: error: unknown type name 'time64_t'
 time64_t _time64(time64_t *t) TIME_IMPL(t)
                  ^~~~~~~~
make[1]: *** [Makefile:6735: fuzz/client.o] Error 1
make[1]: Leaving directory '/c/libOPENSSL-1.1.1-dev/build'
make: *** [Makefile:133: all] Error 2
```

Environment:
- Windows 10 x64,
- mingw-w64 6.3.0,
- MSYS2 20161025,
- OpenSSL-1.1.1-dev (Git master 93d02986656bf4a3ae145f65afc1ca52d2810950).

It reproduced for mingw-w64 build only, but not MSVC.

Could relate to recent commit 14a6570f318562faf850f842ef3ce56f54ddc890, since not reproduced, if sources reverted to cad3f9dacbd03ea686a26690c8f4e58866d08104.

CC:@kroeckx

Alexander

**Comment by richsalz** [comment]
this will be fixed by #3248 very soon.

**Comment by sav-ix** [comment]
Thank you for a quick response. Just applied patch https://github.com/openssl/openssl/pull/3248/commits/2f4f063d4c6b48fc7383d2d5624102eebc6c76a6, and OpenSSL builds using mingw-w64 finished successfully.

**Comment by richsalz** [comment]
fixed with commit 424aa35 now in master, thanks.
